### ISSUE TITLE
Feature Request: `continue_if_timeout` for forcemerge or alternative

### ISSUE THREAD

**Issue by anandsinghkunwar** [elastic/curator#1459]
I have noticed in my ElasticSearch cluster when using curator, that forcemerge gets blocked on a request and on the background the segments reduce to the desired number. However the connection doesn't close and a timeout exception is triggered which leads to my curator job failing. I'd like to ask for a feature request similar to `continue_if_exception` called `continue_if_timeout`(rename to your liking) where instead of moving on to the next action, it moves on to the next index of the list. 

Alternatively, curator catch the timeout exception and check if the number of segments was reduced for the particular index and not throw an exception if it actually did.

**Comment by anandsinghkunwar** [comment]
`continue_if_timeout` feature could be best effort I feel. Or the checking segments thing could just make the behaviour just a tad bit more reliable, it could just throw an exception otherwise. The forcemerge isn't perfect right now either, this change could just make it a little less not perfect. What say you?

**Comment by anandsinghkunwar** [comment]
Can you elaborate more on the keep-alive strategy?

**Comment by anandsinghkunwar** [comment]
Isn't that the needed for multiple http requests, while forcemerge will be just one request.

**Comment by anandsinghkunwar** [comment]
Keep us posted.

**Comment by passing** [comment]
It would be nice to have the `wait_for_completion` option available for the `forcemerge` action in curator:
https://www.elastic.co/guide/en/elasticsearch/client/curator/current/option_wfc.html
So you could just add the forcemerge task to the queue when you are fine about not knowing that it will ever succeed (But if you otherwise get a timeout, you don't know either)

**Comment by untergeek** [comment]
@passing Unfortunately, for reasons built deeply into Lucene from very early on, and not yet resolved, as of today, it is impossible to have a true `wait_for_completion` option for forcemerge. It is simply not acceptable for Curator to proceed on to other tasks while an incomplete forcemerge operation is ongoing. Until there's a proper, repeatable and reliable way to detect current forcemerges (i.e. the tasks API), this remains impossible to code around with any degree of certainty.
